cmd/clusterctl/client/repository/repository_github_test.go

```diff
@@ -30,11 +30,204 @@ import (
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/internal/test"
 )
 
-//TODO: test newGitHubRepository
+func Test_githubRepository_newGitHubRepository(t *testing.T) {
+	g := NewWithT(t)
+
+	type field struct {
+		providerConfig config.Provider
+		variableClient config.VariablesClient
+	}
+	tests := []struct {
+		name    string
+		field   field
+		want    *gitHubRepository
+		wantErr bool
+	}{
+		{
+			name: "can create a new GitHub repo",
+			field: field{
+				providerConfig: config.NewProvider("test", "https://github.com/o/r1/releases/v0.4.1/path", clusterctlv1.CoreProviderType),
+				variableClient: test.NewFakeVariableClient(),
+			},
+			want: &gitHubRepository{
+				providerConfig: config.NewProvider("test", "https://github.com/o/r1/releases/v0.4.1/path", clusterctlv1.CoreProviderType),
+
+				configVariablesClient:    test.NewFakeVariableClient(),
+				authenticatingHTTPClient: nil,
+				owner:                    "o",
+				repository:               "r1",
+				defaultVersion:           "v0.4.1",
+				rootPath:                 ".",
+				componentsPath:           "path",
+				injectClient:             nil,
+			},
+			wantErr: false,
+		},
+		{
+			name: "missing variableClient",
+			field: field{
+				providerConfig: config.NewProvider("test", "https://github.com/o/r1/releases/v0.4.1/path", clusterctlv1.CoreProviderType),
+				variableClient: nil,
+			},
+			want:    &gitHubRepository{},
```

Comment:
I think in these cases we want a `nil`. I know we aren't necessarily asserting on this but as part of documentation for those reading the tests it would make sense to have this as `nil`.